app/course/services/courseStateService.js

```diff
@@ -433,6 +458,13 @@ class CourseStateService {
           case ActionTypes.FETCH_SECTIONS:
             uiState.sectionsFetchInProgress = false;
             return uiState;
+          case ActionTypes.CREATE_SECTION:
+          case ActionTypes.REMOVE_SECTION:
+            var sectionGroups = Object.values(this._state.sectionGroups.list);
```

Comment:
While technically possible, you should avoid accessing the state object from a reducer method. In a proper redux pattern it wouldn't be possible, and accessing the state directly makes it easier to accidentally do very unexpected things. The uiState reducer is expected to only modify and access uiState data, plus whatever was handed to it.

Additionally, I would move the logic and calculate the 'requiresAttention' in the actionCreator itself.

ActionCreators responsibility is preparing data for the reducer,
Reducers responsibility is performing the surgery on the state object.